src/orders-update.js

```diff
@@ -64,38 +64,51 @@ export default class OrdersUpdate {
       })
   }
 
-  // Get the state ID from the API based on the key
-  // getStateReference :: String -> Promise -> String
-  getStateReference (key) {
+  // Get the ID from the API based on the key
+  // getStateReference :: (String, String, String) -> Promise -> String
+  getIdFromKey (key, typeId, endpoint) {
+    if (typeof key === 'object')
```

Comment:
But this function should return the ID (string) not the object.
If we want to handle this case I'd say it belongs in `expandReferences`, this function shouldn't have logic inside imho.